nasa-pictures-stealer/src/main/java/nasapicturesstealer/infrastructure/repository/NasaPicturesRepository.java

```diff
@@ -0,0 +1,72 @@
+package nasapicturesstealer.infrastructure.repository;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import lombok.RequiredArgsConstructor;
+import nasapicturesstealer.domain.entity.NasaCamera;
+import nasapicturesstealer.domain.entity.NasaPicture;
+import nasapicturesstealer.infrastructure.entity.NasaCameraEntity;
+import nasapicturesstealer.infrastructure.entity.NasaPictureEntity;
+import nasapicturesstealer.infrastructure.mapper.NasaCameraMapper;
+import nasapicturesstealer.infrastructure.mapper.NasaPictureMapper;
+import org.springframework.data.util.Pair;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class NasaPicturesRepository {
+
+    private final NasaPictureEntityRepository nasaPictureEntityRepository;
+  
+    private final NasaCameraEntityRepository nasaCameraEntityRepository;
+    
+    private final NasaPictureMapper nasaPictureMapper;
+    
+    private final NasaCameraMapper nasaCameraMapper;
+    
+    public List<NasaPicture> saveDomainEntities(List<NasaPicture> pictures) {
```

Comment:
Hi! Maybe it is good to put @Transactional annotation as there are several calls to database in the method ?